example/generators/event_gen.cpp

```diff
@@ -0,0 +1,179 @@
+using arb::cell_lid_type;
+using arb::cell_size_type;
+using arb::cell_member_type;
+using arb::cell_kind;
+using arb::time_type;
+using arb::cell_probe_address;
+
+// Writes voltage trace as a json file.
+void write_trace_json(const arb::trace_data<double>& trace);
+
+class generator_recipe: public arb::recipe {
+public:
+    // There is just the one cell in the model
+    cell_size_type num_cells() const override {
+        return 1;
+    }
+
+    // Create cell with just a single compartment for the soma:
+    //    soma diameter: 18.8 µm
+    //    mechanisms: pas [default params]
+    //    bulk resistivitiy: 100 Ω·cm [default]
+    //    capacitance: 0.01 F/m² [default]
+    //    synapses: 1 * expsyn
+    arb::util::unique_any get_cell_description(cell_gid_type gid) const override {
+        arb::cell c;
+
+        c.add_soma(18.8/2.0);
+        c.soma()->add_mechanism("pas");
+
+        // Add one synapse at the soma.
+        // This synapse will be the target for all events, from both
+        // event_generators.
+        auto syn_spec = arb::mechanism_spec("expsyn");
+        c.add_synapse({0, 0.5}, syn_spec);
+
+        return std::move(c);
+    }
+
+    cell_kind get_cell_kind(cell_gid_type gid) const override {
+        EXPECTS(gid==0); // There is only one cell in the model
+        return cell_kind::cable1d_neuron;
+    }
+
+    // The cell has one target synapse, which receives both inhibitory and exchitatory inputs.
+    cell_size_type num_targets(cell_gid_type gid) const override {
+        EXPECTS(gid==0); // There is only one cell in the model
+        return 1;
+    }
+
+    // Return two generators attached to the one cell.
+    std::vector<arb::event_generator_ptr> event_generators(cell_gid_type gid) const override {
+        EXPECTS(gid==0); // There is only one cell in the model
+
+        using RNG = std::mt19937_64;
+        using pgen = arb::poisson_generator<RNG>;
+
+        auto hz_to_freq = [](double hz) { return hz*1e-3; };
+        time_type t0 = 0;
+        double w_e =  0.001;
+        double w_i = -0.005;
+
+        // Make two event generators.
+        std::vector<arb::event_generator_ptr> gens;
+
+        // Note that each generator gets a different random number generator.
+        // In practice you would use the gid and generator number to seed the
+        // generator, to get unique and reproduceable streams. For this trivial
+        // example with one cell and two generators, the seeds are hard-coded.
+
+        // Add excitatory generator
+        gens.push_back(
+            arb::make_event_generator<pgen>(
+                cell_member_type{0,0}, // Target synapse (gid, local_id).
+                w_e,                   // Weight of events to deliver
+                RNG(29562872),         // Random number generator to use
+                t0,                    // Events start being delivered from this time
+                hz_to_freq(500)));     // 50 Hz average firing rate
```

Comment:
Comment does not fit with the used value

Reply:
Right you are. I somehow lost changes where I set the frequencies more clearly, like in the docs.